### PR TITLE
Add reversible encoding tests

### PR DESCRIPTION
**Pull request by simonjaeger**
Adding integration tests for reversible encoding and fix the pending alarms' test.

### CODE REVIEW

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/BasicIntegrationTests.cs [hunk 1]
```diff
@@ -27,18 +27,31 @@ public class IntegrationTests : PublisherIntegrationTestBase, IClassFixture<OPCU
...
             // Act
             var messages = await ProcessMessagesAsync(publishedNodesFile, new TimeSpan(0, 0, 1), new TimeSpan(0, 2, 0), 1);
 
             // Assert
             Assert.Single(messages);            
             Assert.Equal("i=2253", messages[0].RootElement[0].GetProperty("NodeId").GetString());
-            Assert.Equal("AlarmConditions", messages[0].RootElement[0].GetProperty("DisplayName").GetString());
-            Assert.Equal("http://opcfoundation.org/AlarmCondition#s=1%3aColours%2fEastTank", messages[0].RootElement[0].GetProperty("Value")[1].GetProperty("Value").GetProperty("SourceNode").GetString());
-            Assert.Equal("100", messages[0].RootElement[0].GetProperty("Value")[1].GetProperty("Value").GetProperty("Severity").ToString());          
+            Assert.Equal("PendingAlarms", messages[0].RootElement[0].GetProperty("DisplayName").GetString());
```
Comment (peterbryntesson) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/BasicIntegrationTests.cs:52]: As you say, these paths are not very elegant. Any way of hiding this?

**Review by peterbryntesson — APPROVED** [review]
LGTM

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/BasicIntegrationTests.cs (hunk 1 above)
Comment (simonjaeger) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/BasicIntegrationTests.cs:52]: I looked around. The only way seem to be to serialize it again, and then deserialize it with another library (i.e. Newtonsoft). I didn't like that solution.

**Comment by simonjaeger** [comment]
Looking into the failing build...

**Review by koepalex — APPROVED** [review]

**Pull request merged**